data/campaigns/Eastern_Invasion/scenarios/03_An_Unexpected_Appearance.cfg

```diff
@@ -173,99 +179,254 @@
+            message= _ "Yes. I have history with their leader. I would trust none but myself to defeat him."
         [/message]
     [/event]
 
+    [event]
+        name=die
+
+        [filter]
+            id=Gweddry,Dacyn
+        [/filter]
+
+        [filter_second]
+            side=3
+        [/filter_second]
+
+        [filter_condition]
+            [have_unit]
+                id=Mal-Tar
+            [/have_unit]
+        [/filter_condition]
+
+        [message]
+            speaker=Mal-Tar
+            message= _ "I... I actually did it!"
+        [/message]
+        [message]
+            speaker=Mal-Skraat
+            message= _ "Well well, I’m impressed! Perhaps this apprenticeship has not been wasted on you after all."
+        [/message]
+        [message]
+            speaker=Mal-Kallat
+            message= _ "We will watch your career with great interest..." # star wars reference
+        [/message]
+        [endlevel]
+            result=defeat
+        [/endlevel]
+    [/event]
     [event]
         name=last breath
         [filter]
-            id=Mal-Skraat
+            id=Mal-Tar
         [/filter]
 
+        [message]
+            speaker=Mal-Tar
+            message= _ "Wait, please, not like this!"
+        [/message]
+
+        # add in the herodeaths. Don't do these before, since we want special behavior for the achievement
+        {HERODEATH_GWEDDRY}
+        {HERODEATH_DACYN}
+    [/event]
+    [event]
+        name=die
+        [filter]
+            id=Mal-Tar
+        [/filter]
+
+        [message]
+            speaker=Mal-Kallat
+            message= _ "I told him he should have studied harder."
+        [/message]
         [message]
             speaker=Mal-Skraat
-            message= _ "No! You have defeated me, and can re-enter Wesnoth!"
+            message= _ "Well, sister, I suppose we’ll have to do this ourselves."
         [/message]
+    [/event]
 
-        {VARIABLE first_fork west}
+    [event]
+        name=die
+        [filter]
+            id=Mal-Skraat
+        [/filter]
 
+        [message]
+            speaker=Mal-Kallat
+            message= _ "Skrai, no!! Cursed Wesnothians, he was my precious little brother..."
+        [/message]
+
+        {VARIABLE skraat_dead yes}
+        {VARIABLE fork_04a yes}
         [endlevel]
             result=victory
-            next_scenario=04a_An_Elven_Alliance
+            next_scenario=04a_An_Elven_Interlude
             bonus=yes
             {NEW_GOLD_CARRYOVER 40}
         [/endlevel]
     [/event]
 
     [event]
-        name=last breath
+        name=die
         [filter]
             id=Mal-Kallat
         [/filter]
 
         [message]
             speaker=Mal-Skraat
-            message= _ "You have defeated my brother, but I will follow you and kill you!"
+            message= _ "Kala, no!! Curse you Wesnothians! You may have killed my sister, but I will chase you east and take my revenge!"
```

Comment:
Non-standard double-exclamation marks?

Reply:
My personal opinion is that you should use either 3 or 1, no other numbers permitted. But maybe Wesnoth has a policy of only 1? Not sure.

Reply:
My feeling is the same, although I don't have any justification for it. But that is why I didn't remark on the use of three elsewhere. Although, I do think the use of three should be very sparingly, perhaps only with all-caps text to emphasis the 'loudness'.

Anyhow, I've marked most of the places where only two are used. Usually unnecessarily, in my opinion.

Reply:
I'm open to either 3, 1, or 2.  I felt having several single-exclamation-mark sentences in a row detracted a little from the dialogue (which sentence is emphasized more than the other?), but it's such a subjective thing that I expect any option will be fine.

Reply:
…what? An exclamation mark is not emphasis. If you want emphasis, use bold or italics. An exclamation mark is… exclamation. Shouting. Yelling. Screaming. Loudly declaring. It's not a form of emphasis.

Reply:
That's what I mean - if I shout one thing really loudly, and then say a second thing , the first thing is more emphasized than the second.

The intent was for _"Kala, no"_ to be shouted very loudly, with the subsequent sentences still loud/angry but not to the same extent.  Bold or italics would probably accomplish the same thing, if that's preferred.

Reply:
I guess either bold or triple explanation marks. I think personally I'd just put "no" in bold.